app/concerns/permission_checking.rb

```diff
@@ -36,6 +36,18 @@ def is_non_nasa_draft_approver?(user:, token:)
     end
   end
 
+  def either_non_nasa_user_or_approver?(user:, token:)
+    is_non_nasa_draft_user?(user: user, token: token) || is_non_nasa_draft_approver?(user: user, token: token)
+  end
+
+  def authorize_proposal_all_user_actions(user:, token:)
+    redirect_to manage_collection_proposals_path unless Rails.configuration.proposal_mode && either_non_nasa_user_or_approver?(user: user, token: token)
+  end
+
+  def authorize_public_actions(user:, token:)
```

Comment:
this seems slightly imprecise because the actions aren't 'public' (and because the method isn't doing the authorizing). 

I would move the 'allowed' or 'authorized' to the end again, and then use something that denotes 'both modes' or 'all modes' or 'multi mode', so might end up with something more like `all_modes_action_authorized?`